app/src/androidTest/java/com/github/polypoly/app/map/MapActivityTest.kt

```diff
@@ -0,0 +1,98 @@
+package com.github.polypoly.app.map
+
+import androidx.compose.ui.test.*
+import androidx.compose.ui.test.junit4.createAndroidComposeRule
+import androidx.test.espresso.action.ViewActions.*
+import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.runBlocking
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.osmdroid.views.overlay.Marker
+
+@RunWith(AndroidJUnit4::class)
+class MapActivityTest {
+
+    @get:Rule
+    val composeTestRule = createAndroidComposeRule<MapActivity>()
+
+    @Before
+    fun setUp() {
+        runBlocking { delay(5000) } // TODO: Find a better way to wait for the UI to update
```

Comment:
You can code a `TryOnRepeat` function for that (that checks whether the UI (or other predicate) is satisfied. Also, waiting 5 seconds is super long 😱 !